meson.build

```diff
@@ -80,6 +88,12 @@ else
   z_dep = dummy_dep
 endif
 
+if get_option('enable_debugger') == 'none'
+  curses_dep = dummy_dep
+else
```

Comment:
Just a comment, not looking for changes:

It would be nice to usr the `required: false` syntax for deps:

`opt_dep = dependency('somedep', required : false)`

Avoid the if-elsr dummy, but I'm not sure we can get around it given the dependency itself is a choice
.

Reply:
I tried that and there are slight differences:
```
foo_dep = dependency('foo', 
                     required : get_option('use_foo'),
                     not_found_message : 'hey, foo was not found, you can pass -Duse_foo=false!')
```
When user will *use* `-Duse_foo=false`, it will still check for library foo *and* display the message:

```
$ meson setup -Duse_foo=false build
...
Run-time dependency foo found: NO
message: hey, foo was not found, you can pass -Duse_foo=false!
```
So if the library was present on the system, **we would link it even when user asked to disable the feature**. (with current solution we don't check for dependency at all).

It all makes sense IMHO :)

That would be ok if we wanted to turn the feature on/off depending on what's available on user's system (just like autoconf does right now for e.g. SDL2_net). But I prefer if we were very noisy about disabling "expected" features and very quiet if user decided to disable one. (That's because back before staging, when SDL_sound was a dependency, I was burned - feature was automatically disabled and I thought missing functionality is a bug/regression).

In the future, we'll have an option for replacing this if-else-checks with https://mesonbuild.com/Build-options.html#features (it has tri-state enabled/disabled/auto - but this is available in meson >= 0.47 (and on Ubuntu 18.04 and Debian Stable we have older versions).

I hope in few months the situation will change (with Ubuntu 16.04 going EOL) and we'll be free to use features from newer meson without introducing burden to packagers :)

Reply:
> It all makes sense IMHO :)

meson's doing something then to earn that praise!

>  I prefer if we were very noisy about disabling "expected" features and very quiet if user decided to disable one.

This makes perfect sense as well - users should know if they're build is about to be crippled. 

> when SDL_sound was a dependency, I was burned - feature was automatically disabled and I thought missing functionality is a bug/regression

Put me in the fire too. Even during early days of the CD-DA patch, I was periodically hit by this when testing stand-alone vs. embedded SDL_Sound. 

> In the future, we'll have an option for replacing this if-else-checks

Excellent!  And great timing to move to meson coinciding with 16.04 going EOL.